### DISCUSSION TITLE
foreign_key when using create()

### DISCUSSION THREAD

**Discussion by anchovie91471** [cjmellor/approval#67] [General]
I may not be understanding the documentation, but when I create a new model with:

` $data = $request->validate([`
`            'name' => 'required|string|max:255',`
`            'default' => 'boolean',`
`            'parent_id' => 'integer|nullable',`
`            'team_id' => 'required|integer',`
`            'user_id' => 'required|integer',`
`        ]);`
`        $district = District::create($data);`

The foreign_key fields in the approvals table is empty. My approvals process using Approval::find($id)->approve() seems to work anyway, but I feel like something is wrong.

I've also tried setting the foreign_key to user_id manually with:

`public function getApprovalForeignKeyName(): string`
`{`
`    return 'author_id';`
`}`

**Comment by Temepest74** [comment]
I had the same issue, I had to change the code. Try something like I did here: https://github.com/Temepest74/approval/commit/788a034f2c2aa4bc7e88a2666bbdef96ce5a2c9c

**Reply by cjmellor** [reply]
Sounds like a useful addition if is causing you to have to write your own fixes. Open for this to be PR'd and I can approve it as long as tests pass.

**Reply by Temepest74** [reply]
As soon as I have time, I will. I have quite a bit of changes I would like to add here, such as events and creator_id

**Comment by cjmellor** [comment]
This has been fixed in v2.

The `foreign_key` is now correctly extracted from the payload when using `Model::create()` and stored separately in the approvals table. On `approve()`, it gets restored to the model.

Thanks for flagging this.
